docs/references/ingress-messages.mdx

```diff
@@ -0,0 +1,82 @@
+
+## Query Calls
+Queries don't modify a canister's state and are answered and signed by a single replica. Responses are obtained within milliseconds, but they are typically not signed by the subnet, which implies that a malicious replica or boundary node could arbitrarily modify the response. 
+
+Some applications may have higher authenticity requirements for query responses. For example, if a dapp queries a proposal description that the user then votes on, it may not be acceptable that a single replica or boundary node can tamper with that information, as they could trick users to vote in the attacker's favor. One can consider issuing query calls as update calls instead, see the next section. While that's easy to implement, it makes the calls significantly slower. To avoid this negative performance impact, for some use cases one can sign the data beforehand and provide [certified variables](/docs/current/developer-docs/security/security-best-practices/data-integrity-and-authenticity#certified-variables) in query responses. However, this can be complex to implement, depending on the specific use case. An example where certified variables are used is the [HTTP asset certification](/docs/current/developer-docs/security/security-best-practices/data-integrity-and-authenticity#use-http-asset-certification-and-avoid-serving-your-dapp-through-rawicp0io). 
+
+Error handling for queries is typically simple: on error, one can simply retry until a successful response is received. 
+
+## Update calls
+Updates can modify a canister's state and thus go through consensus, typically taking 1-3 seconds. They return certified responses (signed by the subnet), providing authenticity within the subnet's corruption bounds.
+
+Update calls can be [synchronous](/docs/current/references/ic-interface-spec#http-sync-call-overview) or [asynchronous](/docs/current/references/ic-interface-spec#http-async-call-overview). 
+
+A [synchronous `call`](https://internetcomputer.org/docs/current/references/ic-interface-spec#http-call) follows the "call and await" pattern. In the success case, it maintains the HTTPS connection until a certified response is produced, as illustrated below. 
+
+![Dev containers 1](./_attachments/ingress_sync_call.png)
+
+Compared to the [asynchronous `call`](/docs/current/references/ic-interface-spec#http-async-call) endpoint, where the [`read_state`](/docs/current/references/ic-interface-spec#http-read-state) endpoint is used to poll for the request status and obtain the reply, the synchronous endpoint removes the need to poll for the response in the success case. 
+
+However, a response may not be produced until some implementation-specific timeout is reached. In that case, the synchronous endpoint behaves like the asynchronous one: it returns a `202` response, and the client needs to poll [read_state](/docs/current/references/ic-interface-spec#http-read-state) to get the request status and eventually obtain the reply, as illustrated below.
```

Comment:
Should this be only "call v2" here? There's no async v3 call (which is a bit weird)

Reply:
I'm not sure I understand: on timeout, v3 also returns a 202 (like v2) and in both cases we need to poll read_state, right? What change are you suggesting?

Reply:
Ah, I didn't realize that 202 applied to `v3` as well - sorry for the noise!